airflow/cli/commands/dag_command.py

```diff
@@ -66,64 +69,82 @@ def dag_backfill(args, dag=None):
     if not args.start_date and not args.end_date:
         raise AirflowException("Provide a start_date and/or end_date")
 
-    dag = dag or get_dag(args.subdir, args.dag_id)
+    if not dag:
+        dags = get_dags(args.subdir, dag_id=args.dag_id, use_regex=args.treat_dag_as_regex)
+    else:
+        dags = dag if type(dag) == list else [dag]
+
+    dags.sort(key=lambda d: d.dag_id)
+
+    dag_id_list = "\n".join(str(dag.dag_id) for dag in dags)
+    question = (
+        "You are about to backfill these {count} DAGs:\n{dag_id_list}\n\nAre you sure? [y/n]"
+    ).format(count=len(dags), dag_id_list=dag_id_list)
+    do_it = ask_yesno(question)
```

Comment:
This is new behaviour and might break some peoples workflows (as I know some people script `airflow backfill` on occasion). I think we should only ask then `treat_dag_as_regex` is set.

Reply:
@potiuk That said, why do we need this confirmation/question at all when the we already have a `--dry-run` flag?